services/galley/src/Galley/API/Internal.hs

```diff
@@ -118,6 +119,57 @@ internalAPI =
       <@> featureAPI
       <@> federationAPI
       <@> conversationAPI
+      <@> iEJPDAPI
+
+iEJPDAPI :: API IEJPDAPI GalleyEffects
+iEJPDAPI = mkNamedAPI @"get-conversations-by-user" (callsFed (exposeAnnotations ejpdGetConvInfo))
+
+-- | An unpaginated, internal http interface to `Query.conversationIdsPageFrom`.  Used for
+-- EJPD reports.  Called locally with very little data for each conv, so we don't expect
+-- pagination to ever be needed.
+ejpdGetConvInfo ::
+  forall r p.
+  ( p ~ CassandraPaging,
+    Member ConversationStore r,
+    Member (Error InternalError) r,
+    Member (Input (Local ())) r,
+    Member (Input Env) r,
+    Member (ListItems p ConvId) r,
+    Member (ListItems p (Remote ConvId)) r,
+    Member P.TinyLog r
+  ) =>
+  UserId ->
+  Sem r [EJPDConvInfo]
+ejpdGetConvInfo uid = do
+  luid <- qualifyLocal uid
+  firstPage <- Query.conversationIdsPageFrom luid initialPageRequest
+  getPages luid firstPage
+  where
+    initialPageRequest = mkPageRequest Nothing
+    mkPageRequest = MTP.GetMultiTablePageRequest (toRange (Proxy @1000))
+
+    getPages :: Local UserId -> ConvIdsPage -> Sem r [EJPDConvInfo]
+    getPages luid page = do
+      let convids = MTP.mtpResults page
+          mk :: Data.Conversation -> Maybe EJPDConvInfo
+          mk conv = do
+            let convType = conv.convMetadata.cnvmType
+                ejpdConvInfo = EJPDConvInfo (fromMaybe "n/a" conv.convMetadata.cnvmName) (tUntagged $ qualifyAs luid conv.convId)
+            -- we don't want self conversations as they don't tell us anything about connections
+            -- we don't want connect conversations, because the peer has not responded yet
+            case convType of
+              RegularConv -> Just ejpdConvInfo
+              -- FUTUREWORK(mangoiv): with GHC 9.12 we can refactor this to or-patterns
+              One2OneConv -> Nothing
+              SelfConv -> Nothing
+              ConnectConv -> Nothing
```

Comment:
```suggestion
            case convType of
              RegularConv -> Just ejpdConvInfo
              -- FUTUREWORK(mangoiv): with GHC 9.12 we can refactor this to or-patterns
              One2OneConv -> Nothing -- already covered by connections
              SelfConv -> Nothing -- this doesn't teach us anything
              ConnectConv -> Nothing -- the peer has not responded yet
```

Reply:
This would complicate `FUTUREWORK(mangoiv): with GHC 9.12 we can refactor this to or-patterns`.